raiden/transfer/mediated_transfer/mediation_fee.py

```diff
@@ -59,63 +163,58 @@ def _update_penalty_func(self) -> None:
             x_list, y_list = tuple(zip(*self.imbalance_penalty))
             self._penalty_func = Interpolate(x_list, y_list)
 
-    def calculate_capped_fee(self, fee: FeeAmount) -> FeeAmount:
-        """ Caps `fee` to 0 if negative and `cap_fees` is `True`. """
-        if self.cap_fees and fee < 0:
-            return FeeAmount(0)
-
-        return fee
-
-    def imbalance_fee(self, amount: PaymentWithFeeAmount, balance: Balance) -> FeeAmount:
-        if self._penalty_func:
-            # Total channel balance - node balance = balance (used as x-axis for the penalty)
-            balance = self._penalty_func.x_list[-1] - balance
-            try:
-                return FeeAmount(
-                    round(self._penalty_func(balance + amount) - self._penalty_func(balance))
-                )
-            except ValueError:
-                raise UndefinedMediationFee()
-
-        return FeeAmount(0)
-
-    def fee_payer(self, amount: PaymentWithFeeAmount, balance: Balance) -> FeeAmount:
-        imbalance_fee = self.imbalance_fee(amount=PaymentWithFeeAmount(-amount), balance=balance)
-
-        flat_fee = self.flat
-        prop_fee = int(round(amount * self.proportional / 1e6))
-        return self.calculate_capped_fee(FeeAmount(flat_fee + prop_fee + imbalance_fee))
-
-    def fee_payee(
-        self, amount: PaymentWithFeeAmount, balance: Balance, iterations: int = 2
-    ) -> FeeAmount:
-        def fee_out(imbalance_fee: FeeAmount) -> FeeAmount:
-            return FeeAmount(
-                round(
-                    amount - ((amount - self.flat - imbalance_fee) / (1 + self.proportional / 1e6))
-                )
-            )
+    def fee(self, balance: Balance, amount: float) -> float:
+        assert self._penalty_func
```

Comment:
Can you add an assertion message here, stating that a penalty function should have been given?

Reply:
so ... the change you made here is to allow absence of penalty function with a `0`? @karlb

Reply:
Yes, no penalty func means that this schedule does not include any imbalance fees. I thought making it work was nicer than providing more error messages, even if it's not used right now. Not sure if that was a good idea.
It would be cleaner to always have a penalty func, then I also don't have to add dummy ones, but the current API does not provide the necessary information to always create one in the constructor (total channel capacity is missing). I didn't find a solution that felt right, yet.

Reply:
Thank you